.github/workflows/build-and-publish.yaml

```
@@ -0,0 +1,52 @@
+name: build & publish Docker image
+
+on:
+  push:
+    branches:
+      # TODO: rename master to main
+      - master
+      # TODO: remove before merge
+      - docker
+
+jobs:
+  build-publish:
+    runs-on: ubuntu-latest
+    steps:
+    - name: checkout
+      uses: actions/checkout@v4
+      with:
+        submodules: recursive
+
+    - name: set up Docker buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: log into the GitHub Container Registry
+      uses: docker/login-action@v2
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ github.token }}
+
+    - name: lowercase repo name
+      id: lower-repo
+      uses: ASzc/change-string-case-action@v6
```

It seems like relying on this unpinned "random" third-party dependency is not necessary because [there are various built-in utilities to lower-case a string](https://stackoverflow.com/a/2264537/1072129).

```suggestion
     run: |
        echo "lowercase=$(echo -n '${{ github.repository }}' | tr '[:upper:]' '[:lower:]')" >> $GITHUB_OUTPUT
```